### DISCUSSION TITLE
Add a way to have paginated pages be nested under a sub-route

### DISCUSSION THREAD

**Discussion by edazpotato** [withastro/roadmap#19] [Proposal]
Add a way to have paginated pages be under a sub-route (the Motivation section explains this in more detail).
I'm not sure what exactly this would look like in practise, but Astro should provide some way to do this out of the box because surely this would be a common use-case.

This isn't a proper rfc yet, I'm just looking for feedback on this idea an suggestions on what it could look like if it were implemented into Astro,

## Motivation
In order to have a routing structure like this
```
/posts                                 Page 1 of posts
/posts/page/2                          Page 2 of posts
/posts/2021-11-30-why-astro-is-awsome  An actual post
```
I currently need to have something like this:
```
/posts/index.astro
/posts/slug.astro
/posts/[page].astro
```
which seems like it would work but it has a multitude of issues, mainly revolving around the `paginate` function. I have a pagination component that I use elsewhere in the app that uses the stuff in `page` prop that the `paginate` function generates. I want to use this pagination component for the posts list page, but that didn't work because I can't use the normal pagination function due to it not supporting this use case (I did try modifying `paginate`'s output but that was an absolute nightmare), so I needed to implement my own pagination system in order to support the routing structure I wanted. That wouldn't be too tough normally, but I also needed to generate the `page` prop and make it look identical to the one outputted by the built-in `paginate` function so that the pagination component kept working.
Basically it's a massive, complex mess that took hours to implement, in order to get a relatively simple output.

**Comment by florian-lefebvre** [comment]
I think paginate now support this usecase, see https://docs.astro.build/en/reference/routing-reference/#paginate. You'd have `[...page].astro` for pagination and `[slug].astro` for articles
